### PR TITLE
Persian translation

### PR DESCRIPTION
**Pull request by MasterKia**
Thanks for this amazing add-on.

What are "Dicarded Tab" and "Selected Tab" (in the Custom Theme section)? How can I find them in Firefox?

Some strings are untranslated because I couldn't find the place they're being used.

### CODE REVIEW

**Comment by MasterKia** [comment]
Persian = پارسی

**Comment by asamuzaK** [comment]
Re:  "Dicarded Tab" and "Selected Tab"

Ref [tabs.Tab - Mozilla | MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab)

> `discarded` Optional
>
>    `boolean`. Whether the tab is discarded. A discarded tab is one whose content has been unloaded from memory, but is still visible in the tab strip. Its content gets reloaded the next time it's activated.

> `highlighted`
>
>    `boolean`. Whether the tab is highlighted, i.e. part of the current tab selection. An active tab is always highlighted, but some browsers may allow additional tabs to be highlighted, for example by clicking them while holding Ctrl, Shift or ⌘ Command keys.
>
>    Firefox for Android doesn't support highlighting multiple tabs, and Firefox desktop requires the `browser.tabs.multiselect` preference (enabled by default).

**Comment by asamuzaK** [comment]
For the tab context menus i.e. &Bookmark Tab, refer to [browser/browser/tabContextMenu.ftl](https://pontoon.mozilla.org/fa/firefox/browser/browser/tabContextMenu.ftl/)

Note that `&B` means that key `B` is applied as an access key.
[1320462 - Add ability to set access key to context menu item](https://bugzilla.mozilla.org/show_bug.cgi?id=1320462)

**Comment by MasterKia** [comment]
> Note that `&B` means that key `B` is applied as an access key.

I found this in the German translation:
```json

  "bookmarkAllTabs_menu": {
    "message": "&Lesezeichen für alle Tabs hinzufügen…", 👈🏻👈🏻👈🏻
    "placeholders": {
      "key": {
        "content": "$1",
        "example": "(&B)" 👈🏻👈🏻👈🏻👈🏻👈🏻👈🏻
      }
    }
  },
```

Shouldn't the "example" be `(&L)` as well?

**Comment by MasterKia** [comment]
Done.

**Comment by asamuzaK** [comment]
I don't have time to review now, please wait until this  weekend.
Thank you for your contribution.

**Comment by asamuzaK** [comment]
A quick review.

Your localization does not match native Firefox localization.

For example, in `New Tab` context menu, access key is `د`.
Ref https://pontoon.mozilla.org/fa/firefox/browser/browser/tabContextMenu.ftl/?string=223283
I don't know where to place `&` in RTL, `&د` or `د&`, menu string should be something like
```
زبانهٔ جدی&د
```
or
```
زبانهٔ جدید&
```

In `Bookmark Tab` context menu, access key is `B` even for `fa`.
Ref https://pontoon.mozilla.org/fa/firefox/browser/browser/tabContextMenu.ftl/?string=197231
Maybe context menu string should be like
```
نشانک‌گذاری زبانه &B
```

FYI, you can load temporary add-ons and see changes from [about:debugging](https://firefox-source-docs.mozilla.org/devtools-user/about_colon_debugging/index.html#this-firefox)

**Comment by asamuzaK** [comment]
The fields under "placeholders" not need to be changed.
Placeholders are mainly used in Japanese translations, and also for plurals.

**Comment by MasterKia** [comment]
> For example, in `New Tab` context menu, access key is `د`.

IMG_20220730_094426

It is "ز".


> I don't know where to place `&` in RTL, ``

`&زبانه جدید`




> Your localization does not match native Firefox localization.



> FYI, you can load temporary add-ons and see changes from [about:debugging](https://firefox-source-docs.mozilla.org/devtools-user/about_colon_debugging/index.html#this-firefox)

Did you check for not matching localizations manually or did this method tell you?

**Comment by MasterKia** [comment]
By the way, most accesskey translations in the reference are the same as the English one.
Some even don't have accesskey.

So there is not much to dom

**Comment by asamuzaK** [comment]
What happens if you translate "bookmarkTabs_menu" message to "Translated Bookmark Tabs (&K)..."
Maybe you should reverse `&K` to `K&`

**Comment by asamuzaK** [comment]
Expected output:
image

**Comment by MasterKia** [comment]
Maybe I'm misunderstanding but why does the add-on have to provide the translation for the browser strings?
Can't the add-on just fallback on browser translations?

**Comment by asamuzaK** [comment]
Because sidebar extension can't include native tab context menu items.
This extension is actually the culmination of a completely useless reinvention of the wheel...
So you need context menu for each locale.

[1376251 - Allow sidebar extensions access to native tab context menu](https://bugzilla.mozilla.org/show_bug.cgi?id=1376251)
[1578655 - Follow up 1280347 - Allow extensions to include native (tab) items when using overrideContext](https://bugzilla.mozilla.org/show_bug.cgi?id=1578655)

**Comment by asamuzaK** [comment]
About `&`:
Does "زبانه جدید&" underline `د`?
Does "(K&)" underline `K`?

**Comment by asamuzaK** [comment]
I installed the language pack and tested it briefly.

* Persian character do not seem to work as access key when combined with `&`.
* It is possible to set `$KEY$` placeholder in all context menus.
  However, the access key is limited to the one set in the English locale, so there is a discrepancy with the access key of the native Persian context menu.

What do you think about the discrepancy, is it acceptable?

**Comment by MasterKia** [comment]
> * It is possible to set `$KEY$` placeholder in all context menus.
>       However, the access key is limited to the one set in the English locale,

If Firefox does show the English accesskey to the user then this discrepancy is ok with me.

**Comment by asamuzaK** [comment]
Please fix the following.

Use `$KEY$` placeholder for menu items like `Translated Menu Item $KEY$`.
i.e. for "bookmarkTab_menu": `نشانک‌گذاری زبانه $KEY$`
Note that menu item key ends with `_menu`.

Directly specify access key for menu items below.
"bookmarkTabs_menu" : "نشانک‌گذاری زبانه‌ها (&K)…"
"unpinTabs_menu" : "جداکردن زبانه‌ها (&B)"

**Comment by MasterKia** [comment]
I hope it's acceptable.

File: src/_locales/fa/messages.json [hunk 1]
```diff
@@ -0,0 +1,808 @@
...
+      }
+    }
+  },
+  "closeOtherTabs_menu": {
+    "message": "بستن زبانه های دیگر $KEY$",
+    "description": "Refer to https://pontoon.mozilla.org/[lang]/firefox/browser/browser/tabContextMenu.ftl",
+    "placeholders": {
+      "key": {
+        "content": "$1",
+        "example": "(&ب)"
```
Comment (asamuzaK) [inline src/_locales/fa/messages.json:71]: Nit `(&O)`

File: src/_locales/fa/messages.json [hunk 2]
```diff
@@ -0,0 +1,808 @@
...
+    "message": "بستن $NUM$ زبانه $KEY$",
+    "description": "Refer to https://pontoon.mozilla.org/[lang]/firefox/browser/browser/tabContextMenu.ftl",
+    "placeholders": {
+      "num": {
+        "content": "$1",
+        "example": "2"
+      },
+      "key": {
+        "content": "$2",
+        "example": "(&ز)"
```
Comment (asamuzaK) [inline src/_locales/fa/messages.json:117]: NIt `(&C)`

File: src/_locales/fa/messages.json [hunk 3]
```diff
@@ -0,0 +1,808 @@
...
+  "optionsWarnUserCSS": {
+    "message": "کد CSS نادرست است."
+  },
+  "pinTab_menu": {
+    "message": "سنجاق کردن زبانه $KEY$",
+    "description": "Refer to https://pontoon.mozilla.org/[lang]/firefox/browser/browser/tabContextMenu.ftl",
+    "placeholders": {
+      "key": {
+        "content": "$1",
+        "example": "(&س)"
```
Comment (asamuzaK) [inline src/_locales/fa/messages.json:626]: Nit `(&P)`

File: src/_locales/fa/messages.json [hunk 4]
```diff
@@ -0,0 +1,808 @@
...
+    "message": "زبانه ها را به صورت افقی نشان میدهد. همچنین، میتوانید زبانه ها را گروه بندی کنید و گروه ها را جمع/باز کنید."
+  },
+  "extensionName": {
+    "message": "زبانه های افقی"
+  },
+  "extensionShortName": {
+    "message": "زبانه ها"
+  },
+  "bookmarkAllTabs_menu": {
+    "message": "نشانک‌گذاری همه زبانه‌ها $KEY$",
```
Comment (asamuzaK) [inline src/_locales/fa/messages.json:15]: `$KEY$…`

File: src/_locales/fa/messages.json [hunk 5]
```diff
@@ -0,0 +1,808 @@
...
+      }
+    }
+  },
+  "bookmarkTabs_menu": {
+    "message": "نشانک‌گذاری زبانه‌ها (&K)…",
+    "description": "Refer to https://pontoon.mozilla.org/[lang]/firefox/browser/browser/tabContextMenu.ftl",
+    "placeholders": {
+      "key": {
+        "content": "$1",
+        "example": "(&K)"
```
Comment (asamuzaK) [inline src/_locales/fa/messages.json:51]: Leave it `(&B)`
This is an example when `$KEY$` is used.

File: src/_locales/fa/messages.json [hunk 6]
```diff
@@ -0,0 +1,808 @@
...
+      }
+    }
+  },
+  "unpinTabs_menu": {
+    "message": "حذف سنجاق زبانه‌ها (&B)",
+    "description": "Refer to https://pontoon.mozilla.org/[lang]/firefox/browser/browser/tabContextMenu.ftl",
+    "placeholders": {
+      "key": {
+        "content": "$1",
+        "example": "(&B)"
```
Comment (asamuzaK) [inline src/_locales/fa/messages.json:804]: Leave it `(&P)`

**Comment by MasterKia** [comment]
Sorry and thanks for pointing those mistakes out.

**Comment by asamuzaK** [comment]
Thanks for your contribution and very quick updates.

**Comment by asamuzaK** [comment]
Try build [sidebarTabs/dest at master · asamuzaK/sidebarTabs](https://github.com/asamuzaK/sidebarTabs/tree/master/dest)

**Comment by MasterKia** [comment]
I'll try and let you know.

A side question: What features/benefits does "`svg.context-properties.content.enabled` => true" bring? Other than [improving dark theme](https://github.com/asamuzaK/sidebarTabs/issues/154).

I didn't enable it after installation and I haven't noticed any problems.

**Comment by asamuzaK** [comment]
> What features/benefits does "svg.context-properties.content.enabled => true" bring? Other than https://github.com/asamuzaK/sidebarTabs/issues/154.

Mainly dark theme, but also for some themes like Alpenglow.

**Pull request merged**